api/app/zonal_stats.py

```diff
@@ -335,19 +334,24 @@ def calculate_stats(
     # Add function to calculate overlap percentage.
     add_stats = None
     if intersect_comparison is not None:
+        pixel_area = calculate_pixel_area(geotiff)
 
-        def intersect_percentage(masked) -> float:
-            # Get total number of elements in the boundary.
+        def intersect_pixels(masked) -> float:
+            # Get total number of elements matching our operator in the boundary.
             intersect_operator, intersect_baseline = intersect_comparison  # type: ignore
+            # If total is 0, no need to count. This avoids returning NaN.
             total = masked.count()
-            # Avoid dividing by 0
             if total == 0:
-                return 0
-            percentage = (intersect_operator(masked, intersect_baseline)).sum()
-            return percentage / total
+                return 0.0
+            return float((intersect_operator(masked, intersect_baseline)).sum())
+
+        def intersect_area(masked) -> float:
```

Comment:
You have a lot of `float` values which are not of the same type. You would probably make it easier to follow the code (and avoid bugs) by doing something like:
```python
AreaInSqKmPerPixel = NewType("AreaInSqKmPerPixel", float)

def intersect_area(masked) -> AreaInSqKmPerPixel:
    ...
    return AreaInSqKmPerPixel(myvalue)
```

Reply:
it's just SqKm, but I see your point. Adding SqKm and Percentage